services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -8,10 +8,22 @@ import scala.concurrent.{ ExecutionContext, Future }
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
-    repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
-      .map(_.toString)
+  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] = {
+    def updateConfiguration(configuration: ProjectConfiguration) = {
+      repository
+        .updateOne(toDocument(configuration), "projectId", projectConfiguration.projectId.value)
+        .map(_.toString)
+    }
+
+    getById(projectConfiguration.projectId) flatMap {
+      case Some(config) if projectConfiguration.projectFileConfigurations != config.projectFileConfigurations =>
+        val oldConfig = config.projectFileConfigurations diff projectConfiguration.projectFileConfigurations
+        val newConfig = projectConfiguration
+          .copy(projectFileConfigurations = oldConfig ++ projectConfiguration.projectFileConfigurations)
```

Comment:
```suggestion
       val newFileConfigurations = projectConfiguration.projectFileConfigurations ++ config.projectFileConfigurations)
        val newConfig = projectConfiguration.copy(projectFileConfigurations = newFileConfigurations)
```
Although this line most likely will be changed because of problems with `equals` / `hashcode` definitions

Reply:
Changed realization.